src/include/migraphx/fileutils.hpp

```diff
@@ -0,0 +1,52 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#ifndef MIGRAPHX_GUARD_MIGRAPHLIB_FILEUTILS_HPP
+#define MIGRAPHX_GUARD_MIGRAPHLIB_FILEUTILS_HPP
+
+#include <migraphx/filesystem.hpp>
+#include <string_view>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+MIGRAPHX_EXPORT fs::path make_executable_filename(std::string_view name);
+MIGRAPHX_EXPORT fs::path make_shared_object_filename(std::string_view name);
+MIGRAPHX_EXPORT fs::path make_object_file_filename(std::string_view name);
+MIGRAPHX_EXPORT fs::path make_static_library_filename(std::string_view name);
+
+inline std::string operator+(std::string_view l, const fs::path& r)
+{
+    return std::string{l} + r.string();
+}
```

Comment:
Can we remove this overload?

Reply:
Can we keep it? Future PRs not yet submitted will change from `std::string` to `fs::path` for filenames (needed for unified path handling between Linux and Windows - `std::string` does not do the job). If the overload is missing, there will be many more changes, e.g., when logging from `MIGRAPHX_...("..." + filename);` to `... + filename.string()`. Furthermore, newer C++ standards have this overload implemented.

Reply:
Can you then add an overload for `operator+(const fs::path& r, std::string l)` as well?